tests/test_keys/test_key_fragments.py

```diff
@@ -19,20 +20,103 @@ def test_kfrag_serialization(alices_keys):
     assert new_frag.bn_sig2 == kfrags[0].bn_sig2
 
 
-def test_cfrag_serialization(alices_keys):
+def test_cfrag_serialization_with_proof_and_metadata(alices_keys):
     priv_key_alice, pub_key_alice = alices_keys
 
     _unused_key, capsule = pre._encapsulate(pub_key_alice.point_key)
-    k_frags = pre.split_rekey(priv_key_alice, pub_key_alice, 1, 2)
+    kfrags = pre.split_rekey(priv_key_alice, pub_key_alice, 1, 2)
+
+    # Example of potential metadata to describe the re-encryption request
+    metadata = { 'ursula_id' : 42, 
+                 'timestamp' : time.time(), 
+                 'capsule' : bytes(capsule), 
+               }
+    metadata = str(metadata).encode()
```

Comment:
Obviously, we want a better form of serialization than just turning a dict into a string.

I presume this is being blocked due to not having a format for metadata in the KMS?

Reply:
Well, to me this is just a rough example for testing purposes. We want something reasonable, beyond a simple `b"this is the metadata of a request"`, but without being something too complex.

Reply:
OK, let's go the the initial approach then.